src/lib.rs

```diff
@@ -0,0 +1,1040 @@
+                return Err(anyhow!(format!(
+                    "Invalid seccomp profile types: {}",
+                    &seccomp_profile.type_
+                )));
+            }
+            if seccomp_profile.type_ == "Localhost" {
+                if let Some(localhost_profile) = &seccomp_profile.localhost_profile {
+                    if !settings.localhost_profiles.contains(localhost_profile) {
+                        return Err(anyhow!(format!(
+                            "Invalid seccomp profile: {}",
+                            &localhost_profile
+                        )));
+                    }
+                } else {
+                    return Err(anyhow!(
+                        "The localhost seccomp profile must be set.".to_string(),
+                    ));
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+/* Do the pod validations. Checks pod annotations and containers Seccomp
+ * security context configurations.
+ */
+fn do_validate(pod: &apicore::Pod, settings: &settings::Settings) -> Result<PolicyResponse> {
+    let annotations_validation = do_validate_annotations(pod, settings);
+    if annotations_validation.is_err() {
+        if let Some(error_message) = annotations_validation.err() {
+            return Ok(PolicyResponse::Reject(error_message.to_string()));
+        }
+    }
+    if let Some(podspec) = &pod.spec {
+        let podpec_validation = do_validate_pod(podspec, settings);
+        if podpec_validation.is_err() {
+            if let Some(error_message) = podpec_validation.err() {
+                return Ok(PolicyResponse::Reject(error_message.to_string()));
+            }
+        }
+
+        let containers_validation = do_validate_containers(&podspec.containers, settings);
+        if containers_validation.is_err() {
+            if let Some(error_message) = containers_validation.err() {
+                return Ok(PolicyResponse::Reject(error_message.to_string()));
+            }
+        }
+
+        if let Some(init_containers) = &podspec.init_containers {
+            let init_containers_validation = do_validate_containers(init_containers, settings);
+            if init_containers_validation.is_err() {
+                if let Some(error_message) = init_containers_validation.err() {
+                    return Ok(PolicyResponse::Reject(error_message.to_string()));
+                }
+            }
+        }
+
+        if let Some(ephemeral_containers) = &podspec.ephemeral_containers {
+            let ephemeral_containers_validation =
+                do_validate_containers(ephemeral_containers, settings);
+            if ephemeral_containers_validation.is_err() {
+                if let Some(error_message) = ephemeral_containers_validation.err() {
+                    return Ok(PolicyResponse::Reject(error_message.to_string()));
+                }
+            }
+        }
+    }
+    Ok(PolicyResponse::Accept)
```

Comment:
Not specific to this line, but I would try to queue all the errors into a vector and then return only here, at the bottom of the method. By doing that we would be able to report all the violations